src/core/ext/filters/client_channel/http_proxy.cc

```diff
@@ -42,16 +42,19 @@
  * credentials if present in the 'http_proxy' env var, otherwise leaves it
  * unchanged. It is caller's responsibility to gpr_free user_cred.
  */
-static char* get_http_proxy_server(char** user_cred) {
+static char* get_http_proxy_server(const grpc_channel_args* args, char** user_cred) {
   GPR_ASSERT(user_cred != nullptr);
   char* proxy_name = nullptr;
   char** authority_strs = nullptr;
   size_t authority_nstrs;
-  /* Prefer using 'grpc_proxy'. Fallback on 'http_proxy' if it is not set.
+  /* Check channel arguments in case a per channel proxy is set. Otherwise,
```

Comment:
Please re-word this comment to something like this:

```
We check the following places to determine the HTTP proxy to use, stopping at the first one
that is set:
1. GRPC_ARG_HTTP_PROXY channel arg
2. grpc_proxy environment variable
3. https_proxy environment variable
4. http_proxy environment variable
If none of the above are set, then no HTTP proxy will be used.
```

Reply:
Modified comment to suggested text.